src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy

```diff
@@ -76,6 +77,9 @@ class IqPolicyEvaluatorWorkflowStep
     else if (iqApplication instanceof String) {
       this.iqApplication = new SelectedApplication(iqApplication)
     }
+    else if (iqApplication instanceof UninstantiatedDescribable) {
+      this.iqApplication = ((UninstantiatedDescribable) iqApplication).instantiate()
```

Comment:
not sure I follow what's going on here.

Reply:
>not sure I follow what's going on here.
Edit: In particular, when does this become an UninstatiatedDescribable?

I'm not going to claim that I understand it.  This is what the debugger told me was actually being passed in and why it was skipping to the throw exception block.   Jenkins uses UninstantiatedDescribable to convert from pipeline code to actual objects.   Not much documentation on it that I have found.  

https://github.com/jenkinsci/structs-plugin/blob/master/plugin/src/main/java/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.java

Reply:
> This is what the debugger told me was actually being passed in and why it was skipping to the throw exception block. 

Was this something that showed up as a result of the auto-app changes, or was this something that showed up as a result of these class changes (from Groovy to Java) in the PR?